src/components/FormSection/SchedulingSection.tsx

```diff
@@ -7,18 +7,37 @@ import {
   TextField,
   Timepicker,
 } from '@folio/stripes/components';
-import React from 'react';
+import React, { ChangeEvent, useState } from 'react';
 import { Field, useField } from 'react-final-form';
 import { FormattedMessage, useIntl } from 'react-intl';
 import { Weekday } from '../../utils/weekdayUtils';
 import { SchedulingFrequency } from '../../types';
 import useLocaleWeekdays from '../../hooks/useLocaleWeekdays';
 
 export default function SchedulingSection() {
+  const [error, setError] = useState(false);
```

Comment:
React final form handles errors for you, please use that. Also, adding your own onChange is overriding the RFF one, and so changes won't get saved to the form state.

Feel free to reply or dm on slack if you need some pointers on how to leverage RFF